src/rsa/convert_nist_rsa_test_vectors.py

```diff
@@ -197,17 +197,20 @@ def main(fn, test_type, padding_alg):
         e = int(case['e'], 16)
         d = int(case['d'], 16)
 
-        if n.bit_length() // 8 < 2048 // 8:
-            debug("Skipping due to modulus length (too small).", DEBUG)
-            continue
-
         if test_type == 'sign':
+            if n.bit_length() < 2048:
+                debug("Skipping due to modulus length (too small).", DEBUG)
+                continue
             if n.bit_length() > 4096:
                 debug("Skipping due to modulus length (too large).", DEBUG)
                 continue
 
             print_sign_test(case, n, e, d, padding_alg)
         else:
+            legacy = case['SHAAlg'] in ["SHA1", "SHA256"]
+            if (n.bit_length() < 2048 and not legacy) or n.bit_length() < 1024:
```

Comment:
Again, I think we want to use the rounding check `n.bit_length() // 8 < 2048 // 8` instead of checking `bit_length()` exactly, unless somebody can point out why not.

In particular, it would be interesting to know what happens when we have a, say, 1022 bit public key, right?